Core/ALife.Core/Utility/EvoNumbers/ReadOnlyEvoNumber.cs

```diff
@@ -0,0 +1,350 @@
+
+namespace ALife.Core.Utility.EvoNumbers
+{
+    /// <summary>
+    /// A number that can be evolved within a range. This version can only be read, not written to.
+    /// </summary>
+    /// <seealso cref="ALife.Core.Utility.EvoNumbers.IEvoNumber"/>
+    [DebuggerDisplay("{ToString()}")]
+    public struct ReadOnlyEvoNumber : IEvoNumber
+    {
+        /// <summary>
+        /// The actual value
+        /// </summary>
+        [JsonIgnore]
+        private double _value;
+
+        /// <summary>
+        /// The value delta maximum
+        /// </summary>
+        [JsonIgnore]
+        private DeltaBoundedNumber _valueDeltaMaximum;
+
+        /// <summary>
+        /// The value maximum
+        /// </summary>
+        [JsonIgnore]
+        private Numerics.BoundedNumber _valueMaximum;
+
+        /// <summary>
+        /// The value minimum
+        /// </summary>
+        [JsonIgnore]
+        private Numerics.BoundedNumber _valueMinimum;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReadOnlyEvoNumber"/> class.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="originalValueEvolutionDeltaMax">The original value evolution delta maximum.</param>
+        /// <param name="minimumValue">The minimum value.</param>
+        /// <param name="maximumValue">The maximum value.</param>
+        /// <param name="valueMinimumAndMaximumEvolutionDeltaMax">The value minimum and maximum evolution delta maximum.</param>
+        /// <param name="valueDelta">The value delta.</param>
+        [JsonConstructor]
+        public ReadOnlyEvoNumber(double value, double originalValueEvolutionDeltaMax, Numerics.BoundedNumber minimumValue, Numerics.BoundedNumber maximumValue, double valueMinimumAndMaximumEvolutionDeltaMax, DeltaBoundedNumber valueDelta)
+        {
+            OriginalValue = value;
+            _value = value;
+            OriginalValueEvolutionDeltaMax = originalValueEvolutionDeltaMax;
+            _valueMinimum = minimumValue;
+            _valueMaximum = maximumValue;
+            ValueMaximumAndMinimumEvolutionDeltaMax = valueMinimumAndMaximumEvolutionDeltaMax;
+            _valueDeltaMaximum = valueDelta;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReadOnlyEvoNumber"/> class.
+        /// </summary>
+        /// <param name="startValue">The value.</param>
+        /// <param name="evoDeltaMax">The evolution delta maximum.</param>
+        /// <param name="absoluteMinimum">The absolute minimum.</param>
+        /// <param name="hardMax">The absolute maximum.</param>
+        public ReadOnlyEvoNumber(double startValue, double evoDeltaMax, double hardMin, double hardMax) : this(startValue, evoDeltaMax, new BoundedNumber(hardMin, hardMin, hardMax), new BoundedNumber(hardMax, hardMin, hardMax), evoDeltaMax, new DeltaBoundedNumber(evoDeltaMax, evoDeltaMax, evoDeltaMax - 1, evoDeltaMax + 1))
```

Comment:
This should be a particular focus, since this reflects the old ROEvoNumber's constructor, but due to how they work differently now a bit, the actual behavior here may have changed (i.e. might be worth double checking that the numbers all roughly match up to what they should be matching to).